### DISCUSSION TITLE
refinement near a point or set of points

### DISCUSSION THREAD

**Discussion by ragusa** [inducer/meshpy#86] [Q&A] [answered]
Hi,

can someone share how to create a 2d mesh (with backend traingle as mesh generator) so that triangles are smaller near a point or a set of points supplied by the user? I see that the triangle.build() method can take a py function for refinement but I am unsure about the input/output arguments for that refinement function. Thanks in advance!

**Comment by inducer** [comment, marked as answer]
Try this example:

https://github.com/inducer/meshpy/blob/19588442f5f400126b6529c5cc03907ba842215d/examples/test_triangle.py

**Reply by ragusa** [reply]
thanks a lot.
